### ISSUE TITLE
80% melee protection = 80% chance to take 0 damage from a thrown item, 20% to take full damage

### ISSUE THREAD

**Issue by 9600bauds** [d3athrow/vgstation13#10176]
We need #9535 for thrown items

**Comment by Exxion** [comment]
It's worse than that
You actually only have a 64% chance of blocking thrown damage with 80% melee resistance because you roll _twice_ and you're only protected if _both_ rolls are in your favor

**Comment by 9600bauds** [comment]
:clap:

**Comment by Imagebard** [comment]
Okay so what are the actual mechanics supposed to be again?
Roll a dice, if it passes you resist the damage according to the ammo resist %, roll another, if it also passes the damage is completely blocked?

**Comment by PJB3005** [comment]
Bay reworked this recently and they made it so there's a combination of % for full block and % blocked always, it's very neat.

**Comment by Imagebard** [comment]
>Bay

Oi, is it possible to port it, then?

**Comment by PJB3005** [comment]
Yes.

**Comment by Exxion** [comment]
@Imagebard In our current system, it is all-or-nothing
Roll twice and only block anything if both rolls come up positive

Wait never mind I misread your comment

**Comment by PJB3005** [comment]
https://github.com/Baystation12/Baystation12/pull/12658

**Comment by ghost** [comment]
Armour values need to be reworked in general

Radiation armour works in the same way (% chance of it not affecting you, with the other lot being it does affect you)

This means that, if you come into proximity of a radioactive source, because of how quick it'll 'pulse' you, you'll be down quicker than anything if you're not in a radsuit.

**Comment by Shadowmech88** [comment]
I think armor values should offer flat percentage-based damage reduction for all damage sources of that damage type.

**Comment by Imagebard** [comment]
So instead of specific laser/bullet/melee protection they should give brute/burn/toxin protection?
How would toxin protection work, here?

**Comment by Shadowmech88** [comment]
To clarify, by "damage type" I was referring to laser/bullet/melee etc.

**Comment by Imagebard** [comment]
Oh, read that wrong.

**Comment by icantthinkofanameritenow** [comment]
@Shadowmech88

Agreed. A flat protection percentage would make armor actually matter more often than not; the chance to block it outright could be retained for Ablative Armor and bulletproof vests, as well as shields of course.

**Comment by Imagebard** [comment]
>And bulletproof vests

Nah
In unrelated news, I'd much rather have a armor that lowers stamina damage.

**Comment by icantthinkofanameritenow** [comment]
@Imagebard

I meant in terms of outright blocking bullets, not reflecting them like Ablative does with lasers. 

Also, if by stamina you mean stunning, then I don't know if the current code would even allow that. You can't exactly be half-stunned, you know.

**Comment by Imagebard** [comment]
Yeah, stunning with batons and tasers is stamina damage.
And yes I think reducing stun times is possible. Depends really. Does that Chief Engineer hardsuit value work?

**Comment by icantthinkofanameritenow** [comment]
@Imagebard

If it actually reduces stun time, I've never seen it do anything.

 I've never seen the stunning code, but it seems to be all or nothing- either you are stunned or you're not, with the amount of time it stuns you being dependent on the stun method itself.

**Comment by Imagebard** [comment]
>with the amount of time it stuns you being dependent on the stun method itself. 

What
But I'm pretty sure you could reduce it
Are stuns seriously this strong
If I mean if Africa had a way to reduce stun times they would still be kings.

**Comment by icantthinkofanameritenow** [comment]
@Imagebard
Yup. There's no such thing as stamina damage on this server, or none I can identify as such.
